libs/remote_storage/src/local_fs.rs

```diff
@@ -171,23 +162,19 @@ impl RemoteStorage for LocalFs {
             }
         }
 
-        // Note that PathBuf starts_with only considers full path segments, but
+        // Note that Utf8PathBuf starts_with only considers full path segments, but
         // object prefixes are arbitrary strings, so we need the strings for doing
         // starts_with later.
-        let prefix = full_path.to_string_lossy();
+        let prefix = full_path.to_string();
```

Comment:
```suggestion
        let prefix = full_path.as_str();
```

This should still work? Did not test.

Reply:
Done https://github.com/neondatabase/neon/pull/5352/commits/b437516edf8ae9e85cc20ae78b77fb8ee003a981 (+ test)